File: ChartView/3DBarChart/3DBarChartVB/CustomCartesianRenderer.vb
Imports Telerik.Charting
Imports Telerik.WinControls.UI

Public Class CustomCartesianRenderer
    Inherits CartesianRenderer

    Private _summaries As Dictionary(Of Object, Double) = New Dictionary(Of Object, Double)()

    Public Sub New(ByVal area As CartesianArea)
        MyBase.New(area)
    End Sub

    Public ReadOnly Property Summaries As Dictionary(Of Object, Double)
        Get
            Return Me._summaries
        End Get
    End Property

    Protected Overrides Sub Initialize()
        MyBase.Initialize()
        Me._summaries = New Dictionary(Of Object, Double)()

        For Each series As BarSeries In Me.Area.Series

            For Each dp As CategoricalDataPoint In series.DataPoints
                Dim key As Object = dp.Category

                If Not Me.summaries.ContainsKey(key) Then
                    Me.summaries.Add(key, 0)
                End If

                Me.summaries(key) += dp.Value.Value
            Next
        Next

        For i As Integer = 0 To Me.DrawParts.Count - 1
            Dim barPart As BarSeriesDrawPart = TryCast(Me.DrawParts(i), BarSeriesDrawPart)

            If barPart IsNot Nothing Then
                Dim customDrawPart As CustomBarSeriesDrawPart = New CustomBarSeriesDrawPart(CType(barPart.Element, BarSeries), Me)
                customDrawPart.DrawCap = barPart.Element.Name = "Last"
                Me.DrawParts(i) = customDrawPart
            End If
        Next
    End Sub
End Class

File: ChartView/3DBarChart/3DBarChart/CustomCartesianRenderer.cs
using System;
using System.Collections.Generic;
using Telerik.Charting;
using Telerik.WinControls.UI;

namespace _3DBarChart
{
    public class CustomCartesianRenderer : CartesianRenderer
    {
        private Dictionary<object, double> summaries = new Dictionary<object, double>();

        public CustomCartesianRenderer(CartesianArea area)
            : base(area)
        { }

        public Dictionary<object, double> Summaries
        {
            get { return this.summaries; }
        }

        protected override void Initialize()
        {
            base.Initialize();

            this.summaries = new Dictionary<object, double>();

            foreach (BarSeries series in this.Area.Series)
            {
                foreach (CategoricalDataPoint dp in series.DataPoints)
                {
                    object key = dp.Category;

                    if (!this.summaries.ContainsKey(key))
                    {
                        this.summaries.Add(key, 0);
                    }

                    this.summaries[key] += dp.Value.Value;
                }
            }

            for (int i = 0; i < this.DrawParts.Count; i++)
            {
                BarSeriesDrawPart barPart = this.DrawParts[i] as BarSeriesDrawPart;

                if (barPart != null)
                {
                    CustomBarSeriesDrawPart customDrawPart = new CustomBarSeriesDrawPart((BarSeries)barPart.Element, this);
                    customDrawPart.DrawCap = barPart.Element.Name == "Last";

                    this.DrawParts[i] = customDrawPart;
                }
            }
        }
    }
}
